applicationset/services/scm_provider/azure_devops_test.go

```diff
@@ -0,0 +1,287 @@
+package scm_provider
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	azureMock "github.com/argoproj/argo-cd/v2/applicationset/services/scm_provider/azure_devops/git/mocks"
+	"github.com/microsoft/azure-devops-go-api/azuredevops"
+	azureGit "github.com/microsoft/azure-devops-go-api/azuredevops/git"
+)
+
+type uuidHolder struct {
+	uuid uuid.UUID
+	err  error
+}
+
+func getUuidHolder() uuidHolder {
+	uuid, err := uuid.NewUUID()
+	return uuidHolder{uuid: uuid, err: err}
+}
+
+func s(input string) *string {
+	return &input
+}
+
+func TestAzureDevopsRepoHasPath(t *testing.T) {
+	organization := "myorg"
+	teamProject := "myorg_project"
+	repoName := "myorg_project_repo"
+	path := "dir/subdir/item.yaml"
+	branchName := "my/featurebranch"
+
+	ctx := context.TODO()
+	uuidHolder := getUuidHolder()
+
+	testCases := []struct {
+		name             string
+		pathFound        bool
+		azureDevopsError error
+		returnError      bool
+	}{
+		{
+			name:      "repo_has_path_found_returns_true",
```

Comment:
Can you replace the underscores with spaces? It'll be easier to read, and Go will automatically change the name to comply with its conventions.

Reply:
Wasn't aware of this, sorry. 
Replacement done, but with capitalization of some letters. Should they be all lowercase?